runtime/src/accounts_db.rs

```diff
@@ -2330,7 +2332,13 @@ impl<'a> AppendVecScan for ScanState<'a> {
         self.pubkey_to_bin_index -= self.bin_range.start;
 
         let balance = loaded_account.lamports();
-        let loaded_hash = loaded_account.loaded_hash();
+        let mut loaded_hash = loaded_account.loaded_hash();
+        if loaded_hash == Hash::default() || self.config.check_hash {
```

Comment:
@yhchiang-sol @brooksprumo here's prototype 1 of where we would figure out we need to rehash an account because a valid hash was not stored in the storage.